### ISSUE TITLE
Update or remove the change management whitepaper

### ISSUE THREAD

**Issue by pp-mo** [SciTools/iris#3286]
I thought we'd removed this so it is no longer in the built docs.
But it turns out it is still "there", though not linked from anywhere else : it can appear in searches
https://scitools.org.uk/iris/docs/latest/whitepapers/change_management.html

We removed it with Iris 2 / dask development
https://github.com/SciTools/iris/pull/2862
This was because we then found that it promises too much on backwards compatibility.
Basically, enabling "dask behaviour" on a FUTURE was not a practical possibility.

TODO NOW: revise + reinstate **or** remove

**? Comment, anyone ?**

**Comment by mo-g** [comment]
Alternative third option: follow what you did with `change_management_goals.rst` and spin the content into other parts of the documentation. Maybe some of this has already happened? I haven't read the documentation comprehensively since er... 1.7. I'm sorry. I'll put it on the to-do list! :disappointed: 

The documentation it contains seems useful and relevant in and of itself, so I'd support some form of retention if it hasn't already been 'reimplemented'.

**Comment by pp-mo** [comment]
Done with https://github.com/SciTools/iris/pull/4217
